### ISSUE TITLE
dependence on execution character set

### ISSUE THREAD

**Issue by Humm42** [skeeto/pdjson#31]
The README rightly states that only UTF-8 is supported, as it
should be. It notes other libraries’ broken Unicode support and a
focus on correctness in this one. All that seems quite right,
except for one little thing: pdjson depends on the execution
character set being ASCII or the like; e.g., it requires the
constant `'{'` to have the value 0x7b.

That is a fine choice and works for most cases, but is not
maximally portable. To break pdjson, you can play around with
`gcc -fexec-charset`.

To fix, the character constants should be replaced by the integer
values (`0x7b /* { */`). C23 introduces UTF-8 character constants
(`u8'{'` of type `unsigned char`), but I suppose you don’t want to
depend on a C23 compiler.

**Comment by skeeto** [comment]
This is true. Considering exactly this, in some programs I've used numeric 
literals instead of character literals. However, I've decided it's just 
not worth worrying about unusual execution character sets. I draw a 
similar line not worrying about machines where bytes aren't octets.

Thanks for pointing out -fexec-charset. I wasn't aware of this option, and 
it makes for interesting new tests. Though, trying it out now I quickly 
ran into its limitations. It doesn't apply to libc interfaces, and simply 
changing locale is insufficient. It's like an ABI change, and libc would 
also need to be be compiled in the chosen execution character set.

**Comment by boris-kolpackov** [comment]
I've been wading through this area not long ago and I am pretty sure I remember reading that the ASCII characters in any alternative source/execution charset should map to their ASCII values. I don't have the reference off hand, but it's probably here somewhere: https://eel.is/c++draft/lex#charset (this is for C++ but I would be surprised if the C rules differ here).
